asserts/gpgkeypairmgr.go

```diff
@@ -67,39 +68,115 @@ func (gkm *gpgKeypairManager) gpg(input []byte, args ...string) ([]byte, error)
 // Importing keys through the keypair manager interface is not
 // suppored.
 // Main purpose is allowing signing using keys from a GPG setup.
-func NewGPGKeypairManager(homedir string) KeypairManager {
-	return &gpgKeypairManager{
+func NewGPGKeypairManager(homedir string) *GPGKeypairManager {
+	return &GPGKeypairManager{
 		homedir: homedir,
 	}
 }
 
-func (gkm *gpgKeypairManager) Put(authorityID string, privKey PrivateKey) error {
-	// NOTE: we don't need this initially at least and this keypair mgr is not for general arbitrary usage
-	return fmt.Errorf("cannot import private key into GPG keyring")
-}
-
-func (gkm *gpgKeypairManager) Get(authorityID, keyID string) (PrivateKey, error) {
-	out, err := gkm.gpg(nil, "--batch", "--export", "--export-options", "export-minimal,export-clean,no-export-attributes", "0x"+keyID)
+func (gkm *GPGKeypairManager) retrieve(fpr string) (PrivateKey, error) {
+	out, err := gkm.gpg(nil, "--batch", "--export", "--export-options", "export-minimal,export-clean,no-export-attributes", "0x"+fpr)
 	if err != nil {
 		return nil, err
 	}
 	if len(out) == 0 {
-		return nil, fmt.Errorf("cannot find key %q in GPG keyring", keyID)
+		return nil, fmt.Errorf("cannot retrieve key with fingerprint %q in GPG keyring", fpr)
 	}
 
 	pubKeyBuf := bytes.NewBuffer(out)
-	privKey, err := newExtPGPPrivateKey(pubKeyBuf, "GPG", gkm.sign)
+	privKey, err := newExtPGPPrivateKey(pubKeyBuf, "GPG", func(content []byte) ([]byte, error) {
+		return gkm.sign(fpr, content)
+	})
 	if err != nil {
-		return nil, fmt.Errorf("cannot use GPG key %q: %v", keyID, err)
+		return nil, fmt.Errorf("cannot load GPG public key with fingerprint %q: %v", fpr, err)
 	}
-	gotID := privKey.PublicKey().ID()
-	if gotID != keyID {
-		return nil, fmt.Errorf("got wrong key from GPG, expected %q: %s", keyID, gotID)
+	gotFingerprint := privKey.fingerprint()
+	if gotFingerprint != fpr {
+		return nil, fmt.Errorf("got wrong public key from GPG, expected fingerprint %q: %s", fpr, gotFingerprint)
 	}
 	return privKey, nil
 }
 
-func (gkm *gpgKeypairManager) sign(fingerprint string, content []byte) ([]byte, error) {
+// Walk iterates over all the RSA private keys in the local GPG setup calling the provided callback until this returns true for done, or errors.
+func (gkm *GPGKeypairManager) Walk(consider func(privk PrivateKey, fingerprint string) (done bool, err error)) error {
```

Comment:
Can we please keep this hidden?  We probably don't even want to be fiddling with the user's GPG keyring at all per the latest conversations, but it's okay to keep this for a follow up. For the time being, let's please just not increase the surface of this type, and avoid exporting such APIs (fingerprint, the walking function, etc).